### PR TITLE
Add Typescript Definitions

### PR DESCRIPTION
**Pull request by schonfeld**
### Contributing checklist
- [x] Code must follow existing styling conventions
- [x] Added a descriptive commit message

### Solves issue(s)

No issue, yet. This adds support for Typescript definitions, based on this repo's documentation & the Tokbox API docs website.

### CODE REVIEW

**Comment by joekim** [comment]
Great job. Looking forward to seeing this merged.

**Comment by msach22** [comment]
@enricop89 @ggoldens @allensarmiento This would be a good addition to the repo. Please take a look when you get a chance :)

File: @types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+    videoType: VideoSource;
+    connection: Connection;
+    creationTime: string;
+  }
+
+  interface Connection {
+    creationTime: string;
+    data: string;
+    connectionId: string;
+    sessionId?: string;
```
Comment (enricop89) [inline @types/index.d.ts:49]: I think we should remove `sessionId` on the Connection interface.

File: @types/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+     */
+    useTextureViews?: boolean;
+
+    /**
+     * Android only - default is false
+     */
+    isCamera2Capable?: boolean;
+
+    /**
+     * Android only - default is false
```
Comment (enricop89) [inline @types/index.d.ts:118]: Please replace it with:

https://tokbox.com/developer/sdks/js/reference/OT.html#initSession - ipWhitelist

File: @types/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+  interface SessionConnectEvent {
+    sessionId: string;
+    connection: Connection;
+  }
+
+  interface SessionDisconnectEvent {
+    sessionId: string;
+  }
+
+  interface ConnectionCreatedEvent extends Connection {}
```
Comment (enricop89) [inline @types/index.d.ts:20]: and please add sessionId

```
interface ConnectionCreatedEvent extends Connection {
 sessionId: string;
}

```

File: @types/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,417 @@
...
+
+    /**
+     * Sent when another client connects to the session. The connection object represents the client’s connection.
+     */
+    connectionCreated?: CallbackWithParam<ConnectionCreatedEvent, any>;
+
+    /**
+     * Sent when another client disconnects from the session. The connection object represents the connection that the client had to the session.
+     */
+    connectionDestroyed?: CallbackWithParam<ConnectionCreatedEvent, any>;
```
Comment (enricop89) [inline @types/index.d.ts:142]: Should we change the name to `ConnectionDedtroyedEvent`?

File: @types/index.d.ts (hunk 3 above)
Comment (christophby) [inline @types/index.d.ts:20]:
```suggestion
  interface ConnectionCreatedEvent extends Connection {
    sessionId: string;
  }
```

File: @types/index.d.ts (hunk 1 above)
Comment (christophby) [inline @types/index.d.ts:49]:
```suggestion
```

**Comment by enricop89** [comment]
@xyclos thanks for your help. Would you mind reviewing the changes I requested above? cheers

**Review by enricop89 — APPROVED** [review]
LGTM

**Pull request merged**